tutorials/how_to/training_transformer_elmo.md

```diff
@@ -0,0 +1,80 @@
+# Training Transformer ELMo
+
+This document describes how to train and use a transformer-based version of ELMo with `allennlp`. The model is a port of the the one described in [Dissecting Contextual Word Embeddings: Architecture and Representation](https://www.semanticscholar.org/paper/4dc99343fdc57cf974746e9549c6ee56f013cee5) by Peters et al.
+
+## Training
+
+1. Obtain training data from http://www.statmt.org/lm-benchmark/1-billion-word-language-modeling-benchmark-r13output.tar.gz.
+    ```
+    export BIDIRECTIONAL_LM_DATA_PATH='/your/path/to/1-billion-word-language-modeling-benchmark-r13output'
+    export BIDIRECTIONAL_LM_TRAIN_PATH=$BIDIRECTIONAL_LM_DATA_PATH'/training-monolingual.tokenized.shuffled/*'
+    ```
+2. Obtain vocab.
+    ```
+    mkdir /your/path/to/vocabulary
+    export BIDIRECTIONAL_LM_VOCAB_PATH=/your/path/to/vocabulary
+    cd $BIDIRECTIONAL_LM_VOCAB_PATH
+    aws --no-sign-request s3 cp s3://allennlp/models/elmo/vocab-2016-09-10.txt .
+    cat vocab-2016-09-10.txt | sed 's/<UNK>/@@UNKNOWN@@/' > tokens.txt
+    rm vocab-2016-09-10.txt
+    echo '*labels\n*tags' > non_padded_namespaces.txt
+    ```
+3. Run training.
+    ```
+    ulimit -n 4096
+    bin/allennlp train training_config/bidirectional_lm.jsonnet --serialization-dir $PATH_TO_SERIALIZATION_DIR
```

Comment:
I would remove `bin/`.  The `allennlp` command should work directly for users that follow the README instructions (even for source installations).

Reply:
I missed that I needed to set `$PATH_TO_SERIALIZATION_DIR`.  I might instead just change it to `output_dir` or something that would actually run (but could be improved).

Reply:
Well--unfortunately if you don't run this in the `allennlp` directory you get an error `file training_config/bidirectional_lm.jsonnet not found`.  Maybe these configurations should be hosted on s3 instead and use http links between each of them?